### ISSUE TITLE
Some questions about the implementation details of unsupervised contrastive learning. 

### ISSUE THREAD

**Issue by SephiHorse** [princeton-nlp/SimCSE#83]
In equation (1) of the paper, do h_i and h_i^{+} both have gradients (in the computational graph)? Or one of them has gradients and the other is just a constant tensor which is detached from the computational graph, e.g. by .detach() function.

**Comment by gaotianyu1350** [comment]
Hi,

Both of them have gradients. It is a complete end-to-end model.

**Comment by SephiHorse** [comment]
Thx!
